src/main/java/com/snowflake/kafka/connector/internal/ingestsdk/KcStreamingIngestClient.java

```diff
@@ -0,0 +1,158 @@
+   * @param clientName the client name to uniquely identify the client
+   */
+  protected KcStreamingIngestClient(Properties streamingClientProps, String clientName) {
+    try {
+      LOGGER.info("Creating Streaming Client: {}", clientName);
+      this.client =
+          SnowflakeStreamingIngestClientFactory.builder(clientName)
+              .setProperties(streamingClientProps)
+              .build();
+
+      assert this.client != null; // client is final, so never need to do another null check
+      assert this.client.getName().equals(clientName);
+    } catch (SFException ex) {
+      throw new ConnectException(ex);
+    }
+  }
+
+  /**
+   * Creates an ingest sdk OpenChannelRequest and opens the client's channel No exception handling
+   * done, all exceptions will be passed through
+   *
+   * @param channelName the name of the channel to open
+   * @param config config to get the database and schema names for the channel
+   * @param tableName table name of the channel
+   * @return the opened channel
+   */
+  public SnowflakeStreamingIngestChannel openChannel(
+      String channelName, Map<String, String> config, String tableName) {
+    OpenChannelRequest channelRequest =
+        OpenChannelRequest.builder(channelName)
+            .setDBName(config.get(Utils.SF_DATABASE))
+            .setSchemaName(config.get(Utils.SF_SCHEMA))
+            .setTableName(tableName)
+            .setOnErrorOption(OpenChannelRequest.OnErrorOption.CONTINUE)
+            .build();
+    LOGGER.info("Opening a channel with name:{} for table name:{}", channelName, tableName);
+
+    return this.client.openChannel(channelRequest);
+  }
+
+  /**
+   * TODO @rcheng: should add retry here and create() even if sdk retries bc network/load issues?
+   * esp with rowset api later Calls the ingest sdk to close the client sdk Swallows all exceptions
+   * and returns t/f if the client was closed because closing is best effort
+   *
+   * @return if the client was successfully closed
+   */
+  public boolean close() {
+    if (this.client.isClosed()) {
+      LOGGER.debug("Streaming client is already closed or null");
+      return true;
+    }
+
+    LOGGER.info("Closing Streaming Client:{}", this.client.getName());
+
+    try {
+      this.client.close();
+      return true;
+    } catch (Exception e) {
+      String message =
+          e.getMessage() != null && !e.getMessage().isEmpty()
+              ? e.getMessage()
+              : "no error message provided";
+
+      String cause =
+          e.getCause() != null
+                  && e.getCause().getStackTrace() != null
+                  && !Arrays.toString(e.getCause().getStackTrace()).isEmpty()
+              ? Arrays.toString(e.getCause().getStackTrace())
+              : "no cause provided";
+
+      // don't throw an exception because closing the client is best effort
+      // TODO @rcheng: are sfexceptions logged? in streamingIngestClient I only saw telemetry before
```

Comment:
Are SFExceptions or client closure failures already captured in telemetry?

Reply:
Good question: not sure, Toby or Jay probably know the answer to that.

Reply:
I am confused about the comment:
We currently dont send every exceptions caused in client through telemetry API 
Here is the interface: `SnowflakeTelemetryService`

Logging the exception here is sufficient IMO.